nemo/collections/asr/data/audio_to_text.py

```diff
@@ -928,8 +969,19 @@ def get_manifest_sample(self, sample_id):
     def __iter__(self):
         return self._dataset.__iter__()
 
+    def _compute_len(self):
+        if self.shard_manifests and torch.distributed.is_available() and torch.distributed.is_initialized():
+            my_len = torch.tensor(len(self.manifest_processor.collection), dtype=torch.int32).cuda()
+            torch.distributed.all_reduce(my_len)
+            my_len = my_len.int()
+            logging.debug(f'sharded - all reduced manifest len {my_len}')
+        else:
+            my_len = len(self.manifest_processor.collection)
+
+        return my_len
+
     def __len__(self):
-        return len(self.manifest_processor.collection)
+        return self.len
```

Comment:
I would rather this be dynamic instead of cached value .. but I suppose we would recreate dataset anyway so it's ok.

Reply:
yeah the alternative would be to do the collection operation each time ..